### ISSUE TITLE
Improve Documentation of underlying structure

### ISSUE THREAD

**Issue by Globegitter** [artificialio/sane#101]
There is nothing really yet that describes what sane actually does under the hood, when you run `sane new` or when you run `sane up`, `sane g api`  or also when using an addon (e.g. `sane-auth`).

Filling this in would help people to understand sane quicker so they can be more confident about the stack, debug easier and contribute easier.

I think a new section in the documentation (https://github.com/artificialio/sane/blob/gh-pages/docs.md) named `Core` just before `Troubleshooting` would make sense. But am open to any other suggestions.
@dowsanjack

**Comment by IanVS** [comment]
A high-level doc/readme would be good, but I think another thing that might really help is adding good comments into the source code itself, for people who really want to get into the details.  Instead of putting all the info in the main readme, I've also seen projects put readme docs throughout the project.  This has the benefit of giving people an idea what's going on inside each directory.  Just some thoughts/suggestions.  I can probably help somewhat, since I have at least a bit of an idea what sane new, generate, etc. are doing.

**Comment by Globegitter** [comment]
@IanVS yep that is a great idea and if you could add some more documentation that would be appreciated. Just take any of the tasks above. I started out with an example here: https://github.com/artificialio/sane/blob/master/lib/commands/generate.js#L17

**Comment by feitian124** [comment]
i am searching similar thing like www.mean.io but use emberjs instead of angurlarjs, hope this will be the one.

**Comment by Globegitter** [comment]
That is awesome @feitian124 check out our https://gitter.im/artificialio/sane if you have any questions, or just to say hi :)

**Comment by mgenev** [comment]
I have some docs on the generate command:

```
done:
generate command looks for closest Project
Project reads the package json
Project looks for node modules which are sane addons by keyword
discovered add ons get returned to generate command
generate command compares user input against the default and the discovered blueprints
if default - make api, model etc. 
if discovered the blueprints.client and .server methods get executed and give further instruction
the instruction is in the form of a machine pack
dependencies get installed on  server with npm and on client with ember install:addon
templates get copied by matching the add-on’s template folder with the project folder, from then on it’s /client/* and server/*
configs get appended to
```
